src/main/java/org/intellij/sdk/codesync/ProjectOpenCloseListener.java

```diff
@@ -52,6 +52,19 @@ public void projectOpened(@NotNull Project project) {
       return;
     }
 
+    // Populate state
+    StateUtils.populateState(project);
+
+    boolean shouldContinue = ProjectUtils.acquireLock();
```

Comment:
In case lock is acquired by some other IDE instance, we will still open socket connection with auth_only=true so that Auth related errors can be shown in each instance.

Reply:
So basically there will be 2 locks.
1- intellij.lock, for all intellij instances to eithrr send diffs OR only do the auth
2- daemon.lock, across different IDEs to restrict only 1 instance of populateBuffer overall

Reply:
Using the new approach I was able to implement both of these locks using a single file.